### ISSUE TITLE
FreeDesktop integration segfault

### ISSUE THREAD

**Issue by jpathy** [keepassxreboot/keepassxc#3389]
[TIP]:  # ( Provide a general summary of the issue in the title above ^^ )
[TIP]:  # ( DO NOT include screenshots of your actual database! )

## Expected Behavior
[NOTE]: # ( Tell us what you expected to happen )
Shouldn't crash.

## Current Behavior
[NOTE]: # ( Tell us what actually happens )
Crashes.

## Possible Solution
[NOTE]: # ( Not required, but suggest a fix/reason for the bug )
Following is the backtrace when attached to the process.
```
Thread 1 "keepassxc" received signal SIGSEGV, Segmentation fault.
0x0000555f71f1db96 in Group::groupsRecursive(bool) const ()
(gdb) bt
#0  0x0000555f71f1db96 in Group::groupsRecursive(bool) const ()
#1  0x0000555f71fdb687 in EntrySearcher::repeat(Group const*, bool) ()
#2  0x0000555f71fdb912 in EntrySearcher::search(QString const&, Group const*, bool) ()
#3  0x0000555f7205393e in FdoSecrets::Collection::searchItems(QMap<QString, QString> const&) ()
#4  0x0000555f7204a04e in FdoSecrets::Service::searchItems(QMap<QString, QString> const&, QList<FdoSecrets::Item*>&) ()
#5  0x0000555f7206019e in FdoSecrets::ServiceAdaptor::SearchItems(QMap<QString, QString> const&, QList<QDBusObjectPath>&) ()
#6  0x0000555f72039d5e in FdoSecrets::ServiceAdaptor::qt_static_metacall(QObject*, QMetaObject::Call, int, void**) ()
#7  0x0000555f7203aab5 in FdoSecrets::ServiceAdaptor::qt_metacall(QMetaObject::Call, int, void**) ()
```

## Steps to Reproduce
[NOTE]: # ( Provide a link to a live example, or an unambiguous set of steps to )
[NOTE]: # ( reproduce this bug. Include code to reproduce, if relevant )
1. Start keepassxc and unlock.
2. Use gpg-agent to start pinentry program.
3. Watch it segfault.

## Context
[NOTE]: # ( How has this issue affected you? What unique circumstances do you have? )
Unusable FDOSecrets.

## Debug Info
KeePassXC - Version 2.5.0 
Revision - 6ae27fa47b070902eabb44ebab68d471eb68d0e9

Qt 5.12.0
Debugging mode is disabled.

Operating system: NixOS 19.09pre185259.362be9608c3 (Loris)
CPU architecture: x86_64
Kernel: linux 5.1.17

Enabled extensions:
- Auto-Type
- Browser Integration
- SSH Agent
- KeeShare (signed and unsigned sharing)
- YubiKey

Cryptographic libraries:
 libgcrypt 1.8.4

**Comment by droidmonkey** [comment]
Will need more information about the structure of your database. Does the shared group have children? Can you replicate with an example database that mimics your structure?

**Comment by jpathy** [comment]
Here is a simple test db that crashes with secrets enabled:
https://doc-0g-0s-docs.googleusercontent.com/docs/securesc/ha0ro937gcuc7l7deffksulhg5h7mbp1/pt657ae7nppkka6bac43jipaicakcd9t/1563645600000/11711133447397395626/*/1gbo6jVY0kjVsVXuqPwASwXn8jVds7XIm?e=download

**Comment by droidmonkey** [comment]
That link does not work, you need to use the "copy share link" feature.

**Comment by ppawel** [comment]
I can confirm there is a segfault and strange behavior with the secret service integration - sometimes it "forgets" which database group is shared, and all apps like Skype or Vivaldi/Chromium become confused... sometimes it segfaults when trying to unlock the database.

Will try to gather more information and reproduce with a test database.

**Comment by droidmonkey** [comment]
This may all be fixed when 2.5.2 is released.

**Comment by droidmonkey** [comment]
Closing this since several Secret Service crashes were fixed in 2.5.2
